### ISSUE TITLE
[DateRangePicker]: [a11y] missing accessible name for dateRangePicker popover

### ISSUE THREAD

**Issue by I585157** [SAP/ui5-webcomponents#8598]
### Bug Description

When the DateRangePicker with a set accessiblenName property is clicked and the calendar popover is displayed, the popover has no accessible name attribute. Popover with role="dialog" should have accessible name. 
![Screenshot 2024-04-01 at 1 26 06 PM](https://github.com/SAP/ui5-webcomponents/assets/127151629/0e0bacb5-072f-44b8-ac5b-bb3a6bd8a9b2)

### Affected Component

DateRangePicker

### Expected Behaviour

When the accessibleName property is set for the DateRangePicker component, the DateRangePicker popover should also have a set accessible name. 

### Isolated Example

https://sap.github.io/ui5-webcomponents/playground/?path=/docs/main-daterangepicker--date-range-picker-overview

### Steps to Reproduce

1. Navigate to DateRangePicker storybook example https://sap.github.io/ui5-webcomponents/playground/?path=/docs/main-daterangepicker--date-range-picker-overview
2. Set the accessibleName property to a string
3. Open the dateRangePicker calendar popover
4. Notice the popover has no accessible name set
...

### Priority

Medium

### UI5 Web Components Version

1.23.1

### Browser

Chrome

### Operating System

Mac OS

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by dobrinyonkov** [comment]
Hello @SAP/ui5-webcomponents-topic-b,

Could you please have a look into this? It seems that the DatePicker's popover does not have a required accessible name.
Aria 1.2 also mentions something about that - https://www.w3.org/TR/wai-aria-1.2/#dialog

Kind Regards,
Dobrin
